pyquil/paulis.py

```diff
@@ -632,7 +634,7 @@ def __add__(self, other):
         new_sum = PauliSum(new_terms)
         return new_sum.simplify()
 
-    def __radd__(self, other):
+    def __radd__(self, other: Union[PauliDesignator, PauliCoefficientDesignator]) -> 'PauliSum':
```

Comment:
I think the docsting below is wrong (probably copy/pasted from `__add__`), and the type of `other` here is just `Number`, as indicated by the `assert` below. Since we're replacing `Number` with `PauliCoefficientDesignator` elsewhere, I guess do the same here and change this `Union` to just `PauliCoefficientDesignator` (and update the docstring pls).

Reply:
I would leave the docstring to use `Number` to reflect the code usage. Because it is likely that https://github.com/python/mypy/issues/3186 will be fixed in the future and we can use `Number` instead of `PauliCoefficientDesignator`.

Reply:
Leaving as `Number` in the docstring is fine.